ui.apps/test/components/commerce/productteaser/actionsTest.js

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+'use strict';
+
+import ProductTeaser, {
+    LocationAdapter
+} from '../../../../src/main/content/jcr_root/apps/core/cif/components/commerce/productteaser/v1/productteaser/clientlibs/js/actions';
+
+let addToCartAction = `<button data-action="addToCart" data-item-sku="1234"
+                class="button__root_highPriority button__root clickable__root button__filled" type="button">
+            <span class="button__content"><span>Add to Cart</span></span>
+        </button>`;
+
+let seeMoreDetailsAction = `<button data-action="details" data-url="/some/random/url"
+                class="button__root_highPriority button__root clickable__root button__filled" type="button">
+            <span class="button__content"><span>See more details</span></span>
+        </button>`;
+
+let generateTeaserHtml = button => `<div class="item__root" data-cmp-is="productteaser">
+  <a class="item__images" href="#"></a>
+  <a class="item__name" href="#"><span>Sample product</span></a>
+    <div class="item__price">
+        <span> $0.99</span>
+    </div>
+    <div class="productteaser__cta">
+       ${button}
+    </div>      
+</div>`;
+
+describe('ProductTeaser', () => {
+    let pageRoot;
+    let teaserRoot;
+    let mockLocation;
+
+    before(() => {
+        mockLocation = sinon.mock(LocationAdapter);
+
+        let body = document.querySelector('body');
+        pageRoot = document.createElement('div');
+        body.appendChild(pageRoot);
+    });
+
+    after(() => {
+        pageRoot.parentNode.removeChild(pageRoot);
+    });
+
+    it('triggers the cart addition event for the Add To Cart CTA', () => {
+        while (pageRoot.firstChild) {
```

Comment:
The `while` loop appears in both tests. Maybe you can move it into a `beforeEach` block.

Reply:
Ack, I'll do this.